gudpy/core/gudpy_yaml.py

```diff
@@ -33,40 +35,54 @@ def getYamlModule(self):
 
     def parseYaml(self, path):
         self.path = path
-        return self.constructClasses(self.yamlToDict(path))
+        try:
+            parsedYAML = self.constructClasses(self.yamlToDict(path))
+        except YAMLError:
+            # Exception caused by yaml parsing library
+            raise YAMLException("Invalid YAML file")
+        except YAMLException as e:
+            # Exception caused by invalid arguments
+            raise YAMLException(e)
+        else:
+            return parsedYAML
 
     def yamlToDict(self, path):
-        # Decide the encoding
-        import chardet
-        with open(path, 'rb') as fp:
-            encoding = chardet.detect(fp.read())['encoding']
-
         # Read the input stream into our attribute.
-        with open(path, encoding=encoding) as fp:
+        with open(path, encoding=self.yaml.encoding) as fp:
             return self.yaml.load(fp)
 
     def constructClasses(self, yamldict):
         instrument = Instrument()
-        self.maskYAMLDicttoClass(instrument, yamldict["Instrument"])
+        if "Instrument" in yamldict:
+            self.maskYAMLDicttoClass(instrument, yamldict["Instrument"])
         instrument.GudrunInputFileDir = os.path.dirname(
             os.path.abspath(
                 self.path
             )
         )
+
         beam = Beam()
-        self.maskYAMLDicttoClass(beam, yamldict["Beam"])
+        if "Beam" in yamldict:
+            self.maskYAMLDicttoClass(beam, yamldict["Beam"])
+
         components = Components()
-        self.maskYAMLSeqtoClss(components, yamldict["Components"])
+        if "Components" in yamldict:
+            self.maskYAMLSeqtoClss(components, yamldict["Components"])
```

Comment:
Perhaps I'm missing something, but shouldn't this be `maskYAMLSeqtoClass`?
```suggestion
            self.maskYAMLSeqtoClass(components, yamldict["Components"])
```